pkg/provider/resource_project.go

```diff
@@ -80,73 +98,140 @@ func (p *ProjectResource) Schema() *schema.Resource {
 	}
 }
 
-func (p *ProjectResource) Create(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	client := api.BuildAPI(meta).ProjectClient()
-	project_name := d.Get("project_name").(string)
+// Configure adds the provider configured client to the data source.
+func (r *projectResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.client = req.ProviderData.(*api.API).ProjectClient()
+}
+
+type cloudProvider struct {
+	CloudProviderId   string `tfsdk:"cloud_provider_id"`
+	CloudProviderName string `tfsdk:"cloud_provider_name"`
+}
 
-	projectId, err := client.Create(ctx, project_name)
+type Project struct {
+	ID             types.String    `tfsdk:"id"`
+	ProjectID      types.String    `tfsdk:"project_id"`
+	ProjectName    types.String    `tfsdk:"project_name"`
+	UserCount      types.Int64     `tfsdk:"user_count"`
+	ClusterCount   types.Int64     `tfsdk:"cluster_count"`
+	CloudProviders []cloudProvider `tfsdk:"cloud_providers"`
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (p projectResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var config Project
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
+	projectId, err := p.client.Create(ctx, config.ProjectName.ValueString())
 	if err != nil {
-		return fromBigAnimalErr(err)
+		resp.Diagnostics.AddError("Error creating project", "Could not create project, unexpected error: "+err.Error())
+		return
 	}
 
-	d.SetId(projectId)
+	project, err := p.client.Read(ctx, projectId)
+	if err != nil {
+		resp.Diagnostics.AddError("Error reading project", "Could not read project, unexpected error: "+err.Error())
+		return
+	}
 
-	if err := p.read(ctx, d, meta); err != nil {
-		return diag.FromErr(err)
+	config.ID = types.StringValue(project.ProjectId)
+	config.ProjectID = types.StringValue(project.ProjectId)
+	config.ProjectName = types.StringValue(project.ProjectName)
+	config.UserCount = types.Int64Value(int64(project.UserCount))
+	config.ClusterCount = types.Int64Value(int64(project.ClusterCount))
+	config.CloudProviders = []cloudProvider{}
```

Comment:
When we create a project, the API is returning a `set` of cloud_provider values. We shouldn't set the `config.CloudProviders` to an empty list here, but we should read the returned values from the API and set them here.